tensorflow/core/kernels/data/name_utils.cc

```diff
@@ -0,0 +1,28 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "tensorflow/core/kernels/data/name_utils.h"
+
+#include "absl/base/attributes.h"
+
+namespace tensorflow {
+namespace data {
+namespace name_utils {
+
+ABSL_CONST_INIT const char kRangeDatasetOpName[] = "RangeDataset";
```

Comment:
Instead of doing this, you should:

1) Create a `DatasetType` enum
2) Create a `ToString()` method that takes an enum and returns the "base" name. The base name should match the string used in the iterator prefix.
3) Create a `DatasetDebugString()` method that takes an enum (and variadic arguments list of arguments) returns the equivalent `DebugString` version of the name "FooDatasetOp(arg1, arg2, ...)::Dataset"
4) Create a `IteratorPrefix()` method that takes an enum and a prefix for downstream iterator and returns the prefix "Iterator::Map::Range"

Individual dataset would then define simply declare their type at construction time and do not do any string manipulation themselves.